code/modules/food_and_drinks/kitchen_machinery/deep_fryer.dm

```diff
@@ -130,16 +127,43 @@ GLOBAL_LIST_INIT(oilfry_blacklisted_items, typecacheof(list(
 		if(user.grab_state < GRAB_AGGRESSIVE)
 			to_chat(user, span_warning("You need a better grip to do that!"))
 			return
-		var/mob/living/carbon/C = user.pulling
-		user.visible_message(span_danger("[user] dunks [C]'s face in [src]!"))
-		reagents.expose(C, TOUCH)
-		var/permeability = 1 - C.get_permeability_protection(list(HEAD))
-		C.apply_damage(min(30 * permeability, reagents.total_volume), BURN, BODY_ZONE_HEAD)
+		var/mob/living/carbon/dunking_target = user.pulling
+		log_combat(user, dunking_target, "dunked", null, "into [src]")
+		user.visible_message(span_danger("[user] dunks [dunking_target]'s face in [src]!"))
+		reagents.expose(dunking_target, TOUCH)
+		var/permeability = 1 - dunking_target.get_permeability_protection(list(HEAD))
+		var/target_temp = dunking_target.bodytemperature
+		var/cold_multiplier = 1
+		if(target_temp < TCMB + 10) // a tiny bit of leeway
+			dunking_target.visible_message(span_userdanger("[dunking_target] explodes from the entropic difference! Holy fuck!"))
+			dunking_target.gib()
```

Comment:
Shouldn't the head be the only exploding bit here considering it's also only part of the target that's being dunked in the fryer?
On a sidenote, achievement?